app/models/concerns/digital_object/persistence.rb

```diff
@@ -43,6 +43,8 @@ def before_save
     # TODO: rewrite with ActiveRecord::Callbacks
     # To be overridden by subclasses
     mint_and_store_doi(Hyacinth::Datacite::Doi::IDENTIFIER_STATUS[:draft]) if @mint_reserved_doi_before_save || @publish_after_save
+  rescue DataciteErrorResponse, DataciteConnectionError, HyacinthError::DoiExists => e
```

Comment:
I see you raising `Hyacinth::Exceptions::DataciteErrorResponse` and `Hyacinth::Exceptions::DataciteConnectionError` in one part of the code, but not catching that fully namespaced error classes here.  Is this a mistake?  If the tests are all passing, it seems like you may be missing some tests for this update.

Reply:
I did not find existing specs for persistence.rb. Assuming there are no issues, I will create persistence_spec.rb in spec/models/concerns/digital_object/

Reply:
@fcd1 I think it might be useful to talk this over in real time and I sent you a message a couple of hours ago in gchat.  Feel free to reply over there any let me know when you're available.  I don't think that what you need to test here is too bad.  Let's work through it together, and talk through some existing test examples.